### ISSUE TITLE
Running production server (`MIX_ENV=prod`) returns `ERR_EMPTY_RESPONSE`

### ISSUE THREAD

**Issue by lambrospetrou** [phoenixframework/phoenix#3783]
### Environment

* Elixir version (elixir -v): Elixir 1.10.2 (compiled with Erlang/OTP 21)
* Phoenix version (mix deps): 1.5.0
* NodeJS version (node -v): `--no-webpack`
* NPM version (npm -v):
* Operating system: Ubuntu 18.04.4 LTS running inside WSL in Windows 10

### Expected behavior

I created a completely new Phoenix project following the instructions at https://hexdocs.pm/phoenix/up_and_running.html and when I run `mix phx.server` I can open the website with `http://localhost:4000` and it works fine.

The project was created with the `--no-ecto` and `--no-webpack` flags.

I wanted to try out the production build to see its performance difference, and I followed the instructions at <https://hexdocs.pm/phoenix/deployment.html#content>, and after running ` PORT=4001 MIX_ENV=prod mix phx.server` I expected that opening `http:://localhost:4001` was going to hit the production-built server.

### Actual behavior

When I run ` PORT=4001 MIX_ENV=prod mix phx.server` the server outputs that it started successfully but when I visit it the response is empty and there is no error message anywhere in the console output.

Output when starting the server:
```
> PORT=4001 MIX_ENV=prod mix phx.server
20:33:33.243 [info] Running HelloWorldPhoenixWeb.Endpoint with cowboy 2.7.0 at :::4001 (http)
20:33:33.265 [info] Access HelloWorldPhoenixWeb.Endpoint at http://example.com
```

Output when `curl`-ing the server (same when using Chrome, it shows `ERR_EMPTY_RESPONSE`):
```
> curl -vv localhost:4001
* Rebuilt URL to: localhost:4001/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 4001 (#0)
> GET / HTTP/1.1
> Host: localhost:4001
> User-Agent: curl/7.58.0
> Accept: */*
>
* Empty reply from server
* Connection #0 to host localhost left intact
curl: (52) Empty reply from server
```

I also tried changing the `config/prod.exs` to use `localhost` or `127.0.0.1` as the url host but same results.

I also tried following the releases guide at https://hexdocs.pm/phoenix/releases.html#content but the result is exactly the same. The server starts but always returns empty responses.

I searched previous issues and couldn't find anything similar.

Is there any known issue with WSL on Windows that prevents the production server from working or am I missing something to get it working?

Thanks a lot for any help!

ps. I was super eager to try the new LiveView features, but I spent the last 2 hours debugging this...

**Comment by josevalim** [comment]
Hi @lambrospetrou! Go to your config/prod.exs and try to set `config :logger, :level, :debug` to see if you can get more info.

If that is not productive add this to your lib/my_app/endpoint:

```elixir
plug :debug, "1"
...
plug :debug, "2"
...
plug :debug, "3"
```

def debug(conn, data) do
  IO.inspect data
  conn
end
```

where you plug the :debug entry at multiple places and see what comes out of it. If you don't make it even to the endpoint, then there is definitely a lower level issue at play.

Alternatively, remove [this line](https://github.com/phoenixframework/phoenix/blob/master/installer/templates/phx_single/config/prod.secret.exs#L17) to rule out any issue with IPv6.

**Comment by lambrospetrou** [comment]
Hmm, I had to remove/comment the [line you linked](https://github.com/phoenixframework/phoenix/blob/master/installer/templates/phx_single/config/prod.secret.exs#L17) and now it works!

Thanks for the fast response :)

Is this something affecting only WSL on Windows, or should I keep an eye for this in production as well when deployed on a standard Linux host?

**Comment by josevalim** [comment]
It is definitely an issue with WSL. Glad it is addressed though!

**Comment by tielur** [comment]
@josevalim 
I'm getting this exact same symptoms that was originally posted. We are running phoenix in docker on alpine. I added logging to the Phoenix Endpoint like below and no logs show up. I was able to start a separate Plug.Router in the same application and that I can hit. We also removed the ipv6 line that was referenced above as well.

Any suggestions on what I can look for next?

```
  use Phoenix.Endpoint, otp_app: :my_app

  plug :before

  def before(conn, params) do
    IO.inspect("BEFORE!")
    conn
  end
```
